WcaOnRails/app/models/user.rb

```diff
@@ -1289,6 +1289,22 @@ def delegate_role
     }
   end
 
+  def delegate_role_with_extra_metadata
+    {
+      end_date: nil,
+      is_active: true,
+      group: self.region,
+      user: self,
+      metadata: {
+        status: self.delegate_status,
+        location: self.location,
+        first_delegated: self.actually_delegated_competitions.min_by(&:start_date)&.start_date,
+        last_delegated: self.actually_delegated_competitions.max_by(&:start_date)&.start_date,
```

Comment:
Rails has `actually_delegated_competitions.minimum(:start_date)` and `.maximum(:start_date)` to compute these directly on SQL level